src/braket/circuits/gates.py

```diff
@@ -60,7 +60,13 @@
 
 
 class H(Gate):
-    """Hadamard gate."""
+    r"""Hadamard gate.
+
+    Unitary matrix:
+        .. math:: \mathtt{H} = \frac{1}{\sqrt{2}} \begin{bmatrix}
+                1 & 1 \\
+                1 & -1 \end{bmatrix}.
+    """
```

Comment:
[super nit] the rendering of "Unitary matrix:" is in some weird heading style ([link to rendered doc](https://amazon-braket-sdk-python--756.org.readthedocs.build/en/756/_apidoc/braket.circuits.gate.html#braket.circuits.gate.Gate.H), see screenshot below). I wonder if it can show up as plain text instead, maybe by adjusting whitespace in the docstring somehow?
![image](https://github.com/amazon-braket/amazon-braket-sdk-python/assets/3620100/c66331a0-a8ac-40fe-97df-1920e874a2a0)

Reply:
```suggestion
    Unitary matrix:
    
        .. math:: \mathtt{H} = \frac{1}{\sqrt{2}} \begin{bmatrix}
                1 & 1 \\
                1 & -1 \end{bmatrix}.
    """
```

This styling can be removed by adding a space between the blocks of text as with the example above.

Reply:
Would this be a better option?

![image](https://github.com/amazon-braket/amazon-braket-sdk-python/assets/112575584/14ac5ae5-710c-4486-bab3-fb948dbe01e2)

I don't think we have much control over the formatting here.

Reply:
Does @math411's suggestion work?

Reply:
@rmshaffer @math411 The comments are addressed. This is the updated view.

<img width="842" alt="Screenshot 2023-10-28 at 2 48 05 PM" src="https://github.com/amazon-braket/amazon-braket-sdk-python/assets/112575584/854d6933-9f31-41a7-bf85-540359974dbf">

Reply:
awesome!